### PR TITLE
[PWS] Fixed encoding on title and description metadata [Take 2]

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Go to http://url-caster-fr.appspot.com/webui and enter "http://www.blackanddecker.fr/".
Before:
```
{
  "metadata": [
    {
      "description": "BLACK+DECKERâ¢ est le premier fournisseur mondial d'outils Ã©lectriques et d'accessoires.",
      "url": "http://www.blackanddecker.fr/",
      "rank": 1000,
      "icon": "http://www.blackanddecker.fr/favicon.ico",
      "title": "Outils Ã©lectriques & Outils de jardin â BLACK+DECKERâ¢",
      "id": "http://www.blackanddecker.fr/"
    }
  ]
}
```
After:
```
{
  "metadata": [
    {
      "description": "BLACK+DECKER™ est le premier fournisseur mondial d'outils électriques et d'accessoires.",
      "url": "http://www.blackanddecker.fr/",
      "rank": 1000,
      "icon": "http://www.blackanddecker.fr/favicon.ico",
      "title": "Outils électriques & Outils de jardin – BLACK+DECKER™",
      "id": "http://www.blackanddecker.fr/"
    }
  ]
}
```

BUG=#382

### CODE REVIEW

File: web-service/helpers.py [hunk 1]
```diff
@@ -189,12 +196,8 @@ def GetContentEncoding(content):
             encoding = value[0]
 
     if encoding is None:
-        try:
-            encoding = 'utf-8'
-            u_value = unicode(content, 'utf-8')
-        except UnicodeDecodeError:
-            encoding = 'iso-8859-1'
-            u_value = unicode(content, 'iso-8859-1')
+        encoding = 'iso-8859-1'
```
Comment (dinhviethoa) [inline web-service/helpers.py]: It sounds like it might break for web sites encoded using utf-8 that don't advertise the charset.

**Comment by dinhviethoa** [comment]
When I'm looking at the html of http://www.blackanddecker.fr, the charset is advertised.
I'm not sure why the code isn't catching it. We probably need to investigate that first.

```
<?xml version="1.0" encoding="iso-8859-1" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" lang="fr">
<head><meta property="og:image" content="http://www.blackanddecker.co.uk/images/sharelogo.jpg" /><meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
```

**Comment by beaufortfrancois** [comment]
The problem here is the fact that http://www.blackanddecker.fr/ badly advertises its charset (as many other websites do). You can see that by running `curl www.blackanddecker.fr | xxd | less` and read:
```
00001b0: 2b44 4543 4b45 52e2 84a2 2065 7374 206c  +DECKER... est l
```

Here, `e2 84a2` is ™ UTF-8 Unicode. This means charset should be UTF-8 not ISO-8859-1.

By checking first if UTF-8 is the appropriate content encoding, we can deliver properly encoded `description` and `title`  values.

File: web-service/helpers.py (hunk 1 above)
Comment (beaufortfrancois) [inline web-service/helpers.py]: Trying utf-8 encoding is done at the very beginning of `GetContentEncoding` function. That way, if a website is encoded using utf-8 but with a misconfigured charset, we can deliver right content.

File: web-service/helpers.py [hunk 2]
```diff
@@ -189,12 +196,8 @@ def GetContentEncoding(content):
             encoding = value[0]
```
Comment (dinhviethoa) [inline web-service/helpers.py]: Therefore, I think we should try to decode with the encoding passed as parameter and invalidate it if it fails. The fallback on iso-8859-1 should always work.

**Comment by dinhviethoa** [comment]
Ok. I'll add this improvement later then :)

File: web-service/helpers.py (hunk 2 above)
Comment (mmocny) [inline web-service/helpers.py]: Sorry Hoa, I didn't see this comment before pulling.

We can revert, but I tested with the known test cases and this so far seems a strict improvement.  Can we do a subsequent patch if its still not perfect?

Comment (dinhviethoa) [inline web-service/helpers.py]: sounds good to me.

Comment (beaufortfrancois) [inline web-service/helpers.py]: For info, the encoding is not passed as parameter of  `GetContentEncoding`. 
What is passed is the page content which is used to detect the encoding of it.

**Pull request merged**